### PR TITLE
estimate gas a bit better

### PR DESCRIPTION
**Pull request by jellegerbrandy**
fix for #196

### CODE REVIEW

File: src/operation.ts [hunk 1]
```diff
@@ -71,8 +72,28 @@ export function sendTransaction<T>(
...
+      } catch (err) {
+        errToReturn = err
+      }
+      observer.error(errToReturn)
+    }
+    let gas: number
+    if (gasEstimate) {
+      gas = new BN(gasEstimate * 1.1)
+    } else {
+      gas =  new BN(6000000)
```
Comment (leviadam) [inline src/operation.ts]: Why use such a big number?
We don't have any action in that system that requires such an amount of gas.

**Comment by leviadam** [comment]
I'm just wondering, can we just leave the gas limit empty and let the web3 provider set it? 
Instead of asking the web3 provider and waiting for a callback.

File: src/operation.ts (hunk 1 above)
Comment (jellegerbrandy) [reply src/operation.ts]: It's just a "big enough" number. Do we know what the maximum amount of gas is for actions in our system so we can set it lower?

**Comment by jellegerbrandy** [comment]
Metamask does indeed its own estimate, so regarding MM we could also leave it empty and rely on MM to provide a decent value.

But: :-)

* That is metamask. For our tests,we use the web3 provider, and it needs a gas price. And another user of the client library (that is not a browser, or that does not want to use metamask) will need the gas estimate as well. So it is better that we provide it.

* I _believe_ (but I am not sure) that MM will also ask the ethereum node for the estimate (but the docs that I found are not clear on this aspect), so we do not save a callback really.

* the extra `estimateGas` call also has some use. We can catch tx errors without spending gas to have them actually fail...

**Comment by jellegerbrandy** [comment]
Actually, MM's estimate = 1.5 * eth_estimateGas. So it **does** use the node's estimate (and so will do the webrequest even if we don't ).

Perhaps we should also use 1.5 (instead of 1.1) ?

File: src/operation.ts (hunk 1 above)
Comment (leviadam) [reply src/operation.ts]: I'm sure 1M is big enough.

**Comment by leviadam** [comment]
@jellegerbrandy 
I agree. I meant actually the call to the web3, not the call to the node.
We are not sure if there will be a delay with this call, and this can cause for example a delay in opening the MM pop-up, which is not very good for us.

**Comment by jellegerbrandy** [comment]
I'm not sure what you mean. The "call to web3" for estimated gas is just a request to the node for the estimate. 

Also, I don't kow what you mean by "I agree" :-) Agree with the current solution? 

We can also do something like : "if the web3 provider is metamask, then do not call estimategas in the code (but let metamask do it); otherwise call it ourselves". But again, i do not think this will be performance gain, as metamask will have to do the request anyway...

**Comment by leviadam** [comment]
OK. Sounds good :)

**Comment by jellegerbrandy** [comment]
currently on hold trying to figure out the problems with estimating gas as commented on the issue...

**Pull request merged**